### PR TITLE
call up_cxxinitialize from app context (preapp task)

### PR DESCRIPTION
**Pull request by sunghan-chang**
1. apps/system: do cxxinitialize from preapp task
2. apps: remove redundant calling up_cxxinitialize

### CODE REVIEW

**Comment by seinfra** [comment]
Target : [a933c58cb429e9945bfabcf17b8c859a81d6fd75] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [a933c58cb429e9945bfabcf17b8c859a81d6fd75] - Code Rule Check OK.

**Comment by manoharahk** [comment]
In my opinion, 
1. cxxinitialization is a loader job. Since we don't have loader, kernel does it similar to initialising other sections.
2. However, I don't see any problem in moving it to pre-app unless we hit a requirement of initialising (.init) section prior to branch to pre-app. 
3. Memory allocated during cxxinitialization(for global static constructors) will be moved from idle task to pre-app task.

**Comment by manoharahk** [comment]
4. static destructors are attached to task group on-exit callback. I suspect there could be some problem if the pre-app exits.

File: os/kernel/init/os_bringup.c [hunk 1]
```diff
@@ -422,10 +422,6 @@ int os_bringup(void)
 	logm_start();
 #endif
 
-#if !defined(CONFIG_BUILD_KERNEL) && defined(CONFIG_HAVE_CXXINITIALIZE)
```
Comment (btheosam) [inline os/kernel/init/os_bringup.c]: Why is this removed from here?

**Comment by seinfra** [comment]
Target : [a933c58cb429e9945bfabcf17b8c859a81d6fd75] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [a933c58cb429e9945bfabcf17b8c859a81d6fd75] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [a933c58cb429e9945bfabcf17b8c859a81d6fd75] - Code Rule Check (C++) OK.

**Comment by sunghan-chang** [comment]
@manoharahk @sijupunnnoose @sangam-swami @kishore-sn It's the time to discuss this.
In app binary separation case, calling `up_cxxinitialize()` from kernel (os_bringup) is impossible. Where is the proper location to do it?

**Comment by seinfra** [comment]
Target : [bfa1b9d9a1ca08ab3dd2b5ce282ed99b98951446] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [bfa1b9d9a1ca08ab3dd2b5ce282ed99b98951446] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [fc0c38aa31ffc04f29adf15ce5c08b21e6290233] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [fc0c38aa31ffc04f29adf15ce5c08b21e6290233] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [309e7d99e2b67b5bb41e2a1f9ecc29491cf1f413] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [309e7d99e2b67b5bb41e2a1f9ecc29491cf1f413] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [27acbc2bd7f5dc2086ac40c46b8d5e40bd005ca2] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [27acbc2bd7f5dc2086ac40c46b8d5e40bd005ca2] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [56dd192727a1a3313bc437e953d95e2fba16453d] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [56dd192727a1a3313bc437e953d95e2fba16453d] - Code Rule Check (C++) OK.
